merkle-tree/concurrent/src/changelog.rs

```diff
@@ -14,6 +11,8 @@ pub struct ChangelogEntry<const HEIGHT: usize> {
     pub path: [[u8; 32]; HEIGHT],
     // Index.
     pub index: u64,
+    // Sequence number.
+    pub seq: u64,
```

Comment:
why do we need seq here?

Reply:
we know:
- the order of appends, index_in_leaves_array
- the number_of_appends
- the final_seq_number

1. say we have appended 4 leaves -> number_of_appends = 4
2. final_seq_number = 8
3. for the second leaf of the 4 leaves index_in_leaves_array = 1
``seq_second_leaf`` = ``final_seq_number - (number_of_appends - index_in_leaves_array)``

Question:
- do you increment seq before assigning it to the event  changelog?
(if so ``seq_second_leaf`` = ``final_seq_number - (number_of_appends - index_in_leaves_array + 1)``

Reply:
We increment seq after each leaf append (so we do it multiple times when appending a batch). It happens here.

You have a good point. Maybe we could change the signature of `update` and `append_batch` to `(usize, usize)`, where we return two values:

* `first_sequence_number`
* `first_changelog_index`

Which then can be used to request the changelog even at the point we want to emit the event.

Reply:
Done, I removed `seq` from the changelog and did what I mentioned above.

Reply:
thanks!